src/js/popup/popup.js

```diff
@@ -1,802 +1,1350 @@
+            const { premiumSubdomainSet } = await browser.storage.local.get("premiumSubdomainSet");            
+            const isPremiumSubdomainSet = premiumSubdomainSet !== "None";  
+          
+            if (!isPremiumSubdomainSet) {
+              const registerSubdomainButton = document.querySelector(".fx-relay-regsiter-subdomain-button");
+              registerSubdomainButton.classList.remove("is-hidden");
+            } else {
+              const generateCustomMask = document.querySelector(".js-generate-custom-mask");
+              
+              // Show "Generate custom mask" button
+              generateCustomMask.classList.remove("is-hidden");
+
+              generateCustomMask.addEventListener("click", (e) => {
+                e.preventDefault();
+                popup.panel.update("custom");
+              }, false);
+              
+              // Restyle Random Mask button to secondary
+              generateRandomMask.classList.remove("t-primary");
+              generateRandomMask.classList.add("t-secondary");
+            }
+          }
+          
+          generateRandomMask.addEventListener("click", (e) => {
+              popup.events.generateMask(e, "random");
+            }, false);
+          
+
+          // If no masks are created, show onboarding prompt
+          if (masks.length === 0) {
+            const noMasksCreatedPanel = document.querySelector(".fx-relay-no-masks-created");
+            noMasksCreatedPanel.classList.remove("is-hidden");
+          }
+
+          // Build initial list
+          // Note: If premium, buildMasksList runs `popup.panel.masks.search.init()` after completing
+          popup.panel.masks.utilities.buildMasksList();
+        
+
+          // Remove loading state
+          document.body.classList.remove("is-loading");
+
+        },
+        search: {
+          filter: (query)=> {
+            
+            const searchInput = document.querySelector(".fx-relay-masks-search-input");
+            searchInput.classList.add("is-active");
+
+            const maskSearchResults = Array.from(document.querySelectorAll(".fx-relay-mask-list li"));
+
+            maskSearchResults.forEach((maskResult) => {
+              const emailAddress = maskResult.dataset.maskAddress;
+              const label = maskResult.dataset.maskDescription;
+              const usedOn = maskResult.dataset.maskUsedOn;
+              const generated = maskResult.dataset.maskGenerated;
+              
+              // Check search input against any mask name, label or used-on/generated for web details
+              const matchesSearchFilter =
+                emailAddress.toLowerCase().includes(query.toLowerCase()) ||
+                label.toLowerCase().includes(query.toLowerCase()) ||
+                usedOn.toLowerCase().includes(query.toLowerCase()) ||
+                generated.toLowerCase().includes(query.toLowerCase());
+              
+              if (matchesSearchFilter) {
```

Comment:
Suggestion: Wonder if something like this would also work, but might be a little too unreadable.

 ```suggestion
              const matchesSearchFilter =
                [emailAddress, label, usedOn, generated]
                  .map((str) => str.toLowerCase())
                  .filter((str) => str.includes(query.toLowerCase()));

              if (matchesSearchFilter > 0) {
```

Reply:
This doesn't work for (at least for me when I tried it) because of the _or_ `||` operator. If any of these are true, return true. By putting them all together, they all have to be true to return true. Disregarding for now.

Reply:
I think I know why its failing. Must be because the original checks for a `boolean`, here we would check to see if we got a `.length` for `matchesSearchFilter` greater than 0 - which I failed to add to line 378. Bummer! We should look at this sometime though.